fhevm/tee_interpreter.go

```diff
@@ -0,0 +1,215 @@
+package fhevm
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/zama-ai/fhevm-go/fhevm/tfhe"
+	"github.com/zama-ai/fhevm-go/tee"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func doOperationGenericUint64(op string, environment EVMEnvironment, caller common.Address, input []byte, runSpan trace.Span, operator func(uint64, uint64) uint64) ([]byte, error) {
```

Comment:
@lovenoble Have you used generics in Go? I think these 2 functions can be combined into one:

```
func doOperation[T uint64 | bool](
	op string,
	environment EVMEnvironment,
	caller common.Address,
	input []byte,
	runSpan trace.Span,
	operator func(uint64, uint64) T,
) ([]byte, error) {
```